### PR TITLE
adds a feature to get an array of all articles, sorted by newest created

### PR DESCRIPTION
**Pull request by ELR143**

### CODE REVIEW

File: __tests__/news-server.test.js [hunk 1]
```diff
@@ -96,3 +96,35 @@ describe("GET /api/articles/:article_id", () => {
       });
   });
 });
+
+describe("GET /api/articles", () => {
+  test("200: responds with an array of articles sorted by the newest first", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
```
Comment (nc-jonathan-mcguire) [inline __tests__/news-server.test.js:106]: Need to check these are sorted in the specified order - can use jest sorted for this

File: controllers/news-controller.js [hunk 2]
```diff
@@ -28,3 +30,27 @@ exports.getArticleById = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.getAllArticles = (req, res, next) => {
+  const comments = countComments();
+  const articles = selectAllArticles();
+  Promise.all([comments, articles])
+    .then(([comments, articles]) => {
+      const commentReference = comments.reduce((current, comment) => {
```
Comment (nc-jonathan-mcguire) [inline controllers/news-controller.js:39]: Could also do a PSQL join for this

Comment (nc-jonathan-mcguire) [reply controllers/news-controller.js:39]: But either works

File: controllers/news-controller.js [hunk 3]
```diff
@@ -28,3 +30,27 @@ exports.getArticleById = (req, res, next) => {
...
+        current[comment.article_id] = parseInt(comment.count);
+        return current;
+      }, {});
+      console.log(commentReference);
```
Comment (nc-jonathan-mcguire) [inline controllers/news-controller.js]: remove log

File: controllers/news-controller.js [hunk 4]
```diff
@@ -28,3 +30,27 @@ exports.getArticleById = (req, res, next) => {
...
+      const updatedArticle = articles.map((article) => {
+        delete article.body;
+        const commentCount = commentReference[article.article_id];
+        return { ...article, comment_count: commentCount || 0 };
+      });
+      console.log(updatedArticle);
```
Comment (nc-jonathan-mcguire) [inline controllers/news-controller.js]: remove log

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Need to make sure we've tested they're in the correct order.

Put a comment about psql joins, but maybe jyst one to think about for next time

File: __tests__/news-server.test.js [hunk 5]
```diff
@@ -2,6 +2,7 @@ const app = require("../app");
 const request = require("supertest");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
+const sort = require("jest-sorted");
```
Comment (ghost) [inline __tests__/news-server.test.js]: Do you need to require this? Is it being used?

File: package.json [hunk 6]
```diff
@@ -24,6 +24,7 @@
     "husky": "^8.0.2",
     "jest": "^27.5.1",
     "jest-extended": "^2.0.0",
+    "jest-sorted": "^1.0.14",
```
Comment (ghost) [inline package.json:27]: Looks like this config isnt complete

Screenshot 2023-11-21 at 15 18 55

**Review by ghost — APPROVED** [review]
I am going to approve this, however I'd like you to look at the instructions for section 5, it does say that you should be making use of database queries to achieve the comment count.

**Pull request merged**